### PR TITLE
validation: simplify contract version checking and update data following Granite upgrade

### PR DESCRIPTION
**Pull request by geoknee**
Currently, the code which validates standard versions is overly complex. It matches a set of contract semvers against multiple contract releases, and then asserts that the contract release is standard.

Here, I am simplifying the code to just validate directly against the standard contract release. 

This still supports the pattern where the "next' contract release semvers (also bytecodes and implementation addresses) can be added to the registry _before_ governance ratifies it. When ratification occurs, it is just then a one-liner to update the standard release. 

Other changes:
* update contract deployments following the Granite upgrade on OP Mainnet and OP Sepolia. Source for contract addresses: https://www.notion.so/oplabs/Mainnet-FP-Fixes-Deployment-8f641ce6d3094a5f8c6105184f43781b and @Inphi 

TODO 
---
- [x] make it so `just validate "*"` passes on this branch

### CODE REVIEW

File: superchain/superchain.go
Comment (bitwiseguy) [inline superchain/superchain.go:1]: Maybe outside the scope of this pr, related to the cleanup in this pr: is there a way to combine the `ContractBytecodeHashes` and `ContractVersions` structs into one? It seems they are tracking the same list of contracts, so there is a lot of duplicate code with struct fields and toml tags.

Also the comment describing `ContractBytecodeHashes` looks like a copy/paste error: https://github.com/ethereum-optimism/superchain-registry/blob/gk/simpler-version-check/superchain/superchain.go#L343-L345

File: validation/standard/standard-immutables.toml [hunk 1]
```diff
@@ -8,6 +8,13 @@
...
+mips = '{"98950":[{"start":178,"length":32},{"start":1771,"length":32}]}'
```
[the remark this answers is not in the document]
Comment (bitwiseguy) [reply validation/standard/standard-immutables.toml:15]: Do we expect to have to run this manual step every time there is a new standard contracts release? Is it documented in a runbook somewhere?

[the remark this answers is not in the document]
Comment (geoknee) [reply validation/standard/standard-immutables.toml:15]: Currently, yes we would need to run this manual step every time. I think longer term we will move away from using immutables in L2 contracts which would obviate the need for this manual step.

I update this runbook to cover this https://www.notion.so/oplabs/RB-154-How-To-Update-superchain-registry-when-changing-rollup-json-or-genesis-json-or-contract-dep-ad3753864a234779971a42147cca8f14

File: superchain/superchain.go
Comment (geoknee) [reply superchain/superchain.go:1]: Yes there is probably some clean up possible with generics I'll take a look at this in a followup.

**Pull request merged**